File: ECE39595Homeworks/HW4Assigned/userid/ToyDuck.cpp
#include "ToyDuck.h"
#include "Quack.h"
#include "QuackBehavior.h"
#include "FlyNoWay.h"
#include "MuteQuack.h"
#include "LaysToyEggs.h"
#include <iostream>

ToyDuck::ToyDuck() {
    quackBehavior = new Quack();
    flyBehavior = new FlyNoWay();
    laysEggs = new LaysToyEggs();
}

void ToyDuck::display() {
    std::cout << "I am a toy duck." << std::endl;
}

File: ECE39595Homeworks/HW14Assigned/rkothand/ToyDuck.java
public class ToyDuck extends Duck{
        public ToyDuck(){
            flyBehavior = new FlyPoorly();
            quackBehavior = new MuteQuack();
            laysEggsBehavior = new DoesNotLayEggs();

        }

        public void display(){
            System.out.println("I am a toy duck");
        }
}
